### PR TITLE
Tools: Product-specific visual comparison

### PR DESCRIPTION
**Pull request by bre1470**

### CODE REVIEW

**Review by goransle — APPROVED** [review]
Looks good!

**Review by TorsteinHonsi — CHANGES_REQUESTED** [review]
Okay, let's try this 🤞 . But first, there is a conflict to resolve.

**Review by goransle — COMMENTED** [review]
Looks like the Cypress workflow needs the fetch-depth setting as well. Also the dashboards/script is gone from master section of the filesize job?

**Comment by bre1470** [comment]
I moved file size comparison into its own workflow, so that we can keep both, Highcharts and Dashboards, in that regard together while visuals are separate.

File: tools/gulptasks/lib/test.js [hunk 1]
```diff
@@ -17,79 +17,65 @@
...
+    const ChildProcess = require('node:child_process');
+
+    const paths = Array.from(new Set([
+            ...ChildProcess
+                .execSync('git diff --cached --name-only --diff-filter=ACM')
+                .toString()
+                .split('\n')
+                .filter(match => !!match),
+            ...ChildProcess
+                .execSync('git diff refs/heads/master HEAD --name-only --diff-filter=ACM')
```
Comment (goransle) [inline tools/gulptasks/lib/test.js:36]: Still see some git weirdness here.
image

Locally this worked for me on a single-branch checkout:
```
➜  shallowcharts git:(tools/product-visual-ci) git remote set-branches origin 'master'
➜  shallowcharts git:(tools/product-visual-ci) git fetch -v --depth=1
➜  shallowcharts git:(tools/product-visual-ci) git diff HEAD origin/master
```

**Pull request closed without merging**